File: Back/.Net/BetFriends.Api/Features/Authentication/SignInController.cs
using BetFriends.Users.Application.Abstractions;
using BetFriends.Users.Application.Features.SignIn;
using Microsoft.AspNetCore.Mvc;
using Swashbuckle.AspNetCore.Annotations;

namespace BetFriends.Api.Features.Authentication;

[Route("auth")]
public class SignInController(IUserModule userModule) : Controller
{
    private readonly IUserModule userModule = userModule;

    [SwaggerOperation(Tags = new[] { "Authentication" })]
    [HttpPost]
    public async Task<IActionResult> SignInAsync([FromBody] SignInRequest signInRequest)
    {
        var request = new SignInQuery(signInRequest.Email, signInRequest.Password);
        var authentication = await userModule.ExecuteAsync(request);
        if(authentication != null)
            return Ok(authentication);
        return BadRequest("Erreur d'authentification");
    }
}

public record SignInRequest(string Email, string Password);

File: Back/NodeJS/api/nest/src/user/features/signIn.controller.ts
import { Body, Controller, Inject, Post, Res } from "@nestjs/common";
import { SignInRequest } from "../../../../../modules/users/src/application/features/sign-in/SignInQuery"
import { FastifyReply } from "fastify";
import { ApiProperty } from "@nestjs/swagger";
import { IUserModule } from "../../../../../modules/users/src/application/abstractions/IUserModule";

export class AuthRequest {
  @ApiProperty()
    email: string;
    @ApiProperty()
    password: string;
}

@Controller('signin')
export class SignInController {
    constructor(@Inject('IUserModule') private readonly userModule: IUserModule) {}

  @Post()
  async create(@Body() input: AuthRequest, @Res() res: FastifyReply) {
    const authentication = await this.userModule.Execute(new SignInRequest(input.email, input.password));
    if(authentication == undefined) {
      return res.code(400).send("Erreur d'identification");
    }
    return res.code(200).send(authentication)
  }
}